Add form originated http redirect task

MikeAlhayek opened the pull request:
Fix #14595

sebastienros commented on src/OrchardCore.Modules/OrchardCore.Forms/Activities/FormOriginatedHttpRedirectTask.cs:
```diff
@@ -0,0 +1,71 @@
+            && _httpContextAccessor.HttpContext.Request.Form.TryGetValue(FormPart.RequestOriginatedFromInputName, out var value)
+            && !string.IsNullOrWhiteSpace(value))
+        {
+            WorkflowHttpRedirectResult.Instance.Url = GetLocationUrl(value.ToString());
```
Can't use a static field to store transient state.

sebastienros commented on src/OrchardCore.Modules/OrchardCore.Workflows/Http/Controllers/HttpWorkflowController.cs:
```diff
@@ -276,6 +276,11 @@ private IActionResult GetWorkflowActionResult()
                 return new EmptyResult();
             }
 
+            if (HttpContext.Items.TryGetValue(WorkflowHttpRedirectResult.Instance, out var value) && value is WorkflowHttpRedirectResult result)
```
Is there an existing activity to do a redirect to a specific url? Then it could be used with the stored url.

jtkech commented:
There is already an `HttpRedirectTask` activity, could it be used?

MikeAlhayek commented:
@jtkech yes `HttpRedirectTask` can be used when you know the location you want to redirect to. But, if you have a form that is places on many pages like footer, you want to redirect the user back where they submitted the form from.

jtkech commented:
Okay, hmm can the return url be retrieved from the workflow context and then passed to the redirect activity? as I remember we have Get/Set activities and liquid/scripts to retrieve values from the httpContext.

MikeAlhayek commented:
No. Because the form post the the workflow controller, so we don't know where the request came from. So the original source is lost because workflow is triggered from a controller on a different path than where the form came from.

Today, during triage I mentioned saving the form location into the workflow input. This is something I'll work on later on.

jtkech commented:
Okay then

MikeAlhayek commented:
@sebastienros I addressed the feedback from the demo. Now the you can specify a key name per form which is used to get the value from. By default we use `__RequestOriginatedFrom` but the user can change it to anything they want.

MikeAlhayek commented:
@sebastienros I made the requested changes.
